src/app/settings/views/Configuration/Security/SessionTimeout/SessionTimeout.tsx

```diff
@@ -0,0 +1,117 @@
+import { useState } from "react";
+
+import { Spinner, Select, Button, Icon } from "@canonical/react-components";
+import { useDispatch, useSelector } from "react-redux";
+
+import FormikField from "app/base/components/FormikField";
+import FormikForm from "app/base/components/FormikForm";
+import { actions as configActions } from "app/store/config";
+import configSelectors from "app/store/config/selectors";
+
+type SessionTimeoutFormValues = {
+  session_length: number;
+  time_unit: "hours" | "days";
+};
+
+export enum Labels {
+  Loading = "Loading...",
+  Edit = "Edit",
+  Expiration = "Session timeout expiration",
+  Save = "Save",
+  Cancel = "Cancel",
+  TimeUnit = "Time unit",
+  Days = "Day(s)",
+  Hours = "Hour(s)",
+  ConfigureSessionTimeout = "Configure Session Timeout",
+}
+
+const SessionTimeout = (): JSX.Element => {
+  const dispatch = useDispatch();
+  const configLoading = useSelector(configSelectors.loading);
+  const sessionLength = useSelector(configSelectors.sessionLength) || 1209600; // fallback to default value of 14 days if none is set
+  const saved = useSelector(configSelectors.saved);
+  const saving = useSelector(configSelectors.saving);
+
+  const [isFormOpen, setFormOpen] = useState(false);
+
+  const timeUnitOptions = [
+    { label: Labels.Hours, value: "hours" },
+    { label: Labels.Days, value: "days" },
+  ];
+
+  if (configLoading) {
+    return <Spinner aria-label={Labels.Loading} text={Labels.Loading} />;
+  }
+
+  if (!isFormOpen) {
+    return (
+      <>
+        <span className="u-text--muted">
+          Session timeout expiration
+          <Button
+            className="is-small p-button__show-form"
+            onClick={() => setFormOpen(true)}
+          >
+            <Icon name="edit" />
+            Edit
+          </Button>
+        </span>
+        <p>{sessionLength / 24 / 3600} days</p>
+      </>
+    );
+  } else
+    return (
+      <>
+        <FormikForm<SessionTimeoutFormValues>
+          aria-label={Labels.ConfigureSessionTimeout}
+          buttonsAlign="right"
+          buttonsBordered
+          cleanup={configActions.cleanup}
+          initialValues={{
+            session_length: sessionLength / 24 / 3600,
+            time_unit: "days",
+          }}
+          onCancel={(values, { resetForm }) => {
+            resetForm();
+            values.session_length = sessionLength / 24 / 3600;
+            values.time_unit = "days";
+            setFormOpen(false);
+          }}
+          onSubmit={(values) => {
+            dispatch(configActions.cleanup());
+            dispatch(
+              configActions.update({
+                session_length: Number(
+                  values.time_unit === "hours"
+                    ? values.session_length * 3600
+                    : values.session_length * 3600 * 24
+                ),
```

Comment:
Using [durationToSeconds](https://github.com/ndv99/maas-ui/blob/f92467af7f2fafa39d2a3a406c079056f3c8eaa1/src/app/utils/timeSpan.ts#L22) might be a bit more elegant and clear.
```suggestion
                session_length: durationToSeconds({
                  [values.time_unit]: values.session_length,
                }),
```
If you're not a fan of this you could use util functions such as [hoursToSeconds](https://date-fns.org/v2.29.3/docs/hoursToSeconds) and daysToSeconds to make it less error prone and clearer what type of conversion happens here.

Reply:
That looks good, I'll implement that. Also gonna create a function durationToDays for displaying the values